Cryptic error message when destination folder is not accessible

If the plugin fails to extract the node archive because it can't access the destination directory, the script will produce a somewhat cryptic error:

    [ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.0:install-node-and-npm
    (install node and npm) on project myproject: Could not extract the Node archive:
    Could not extract archive: '/home/user/.m2/repository/com/github/eirslett/node/4.4.5/node-4.4.5-linux-x64.tar.gz':
    No such file or directory

It's thrown by [`DefaultArchiveExtractor.extract():destPath.createNewFile();`](https://github.com/eirslett/frontend-maven-plugin/blob/master/frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/ArchiveExtractor.java#L43) which will produce the cryptic error message that can mislead the debugging process (at least I spent a decent amount of time trying to find out what was wrong with my node archive file)

    throw new ArchiveExtractionException("Could not extract archive: '"
                    + archive
                    + "'", e)


I can quickly add a PR to add a validation for this in the extract method, let me know if you want me to send it in.

--- a/frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/ArchiveExtractor.java
+++ b/frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/ArchiveExtractor.java
@@ -9,6 +9,7 @@ import java.io.File;
 import java.io.FileInputStream;
 import java.io.FileOutputStream;
 import java.io.IOException;
+import java.nio.file.AccessDeniedException;
 
 class ArchiveExtractionException extends Exception {
     ArchiveExtractionException(String message, Throwable cause) {
@@ -39,6 +40,10 @@ final class DefaultArchiveExtractor implements ArchiveExtractor {
                 } else {
                     if (!destPath.getParentFile().exists()) {
                         destPath.getParentFile().mkdirs();
+                    }
+                    if(!destPath.getParentFile().canWrite()) {
+                        throw new AccessDeniedException(
+                                String.format("Could not get write permissions for '%s'", destinationDirectory));
                     }
                     destPath.createNewFile();
                     boolean isExecutable = (tarEntry.getMode() & 0100) > 0;

--- a/pom.xml
+++ b/pom.xml
@@ -60,8 +60,8 @@
                     <artifactId>maven-compiler-plugin</artifactId>
                     <version>3.1</version>
                     <configuration>
-                        <source>1.6</source>
-                        <target>1.6</target>
+                        <source>1.7</source>
+                        <target>1.7</target>
                     </configuration>
                 </plugin>
             </plugins>

